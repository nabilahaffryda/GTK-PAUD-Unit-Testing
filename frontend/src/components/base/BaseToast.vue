<template>
  <v-snackbar top right :timeout="timeout" :color="color" style="opacity: 0.8" v-model="active">
    <v-card flat color="transparent">
      <v-card-text class="pa-0">
        <v-list color="transparent" class="pa-0">
          <v-list-item>
            <v-list-item-icon v-if="icon.length > 0">
              <v-icon x-large dark>{{ icon }}</v-icon>
            </v-list-item-icon>
            <v-list-item-content>
              <span class="white--text body-1" v-html="text"></span>
            </v-list-item-content>
          </v-list-item>
        </v-list>
      </v-card-text>
      <v-divider light></v-divider>
      <v-card-actions>
        <v-spacer></v-spacer>
        <v-btn
          class="text-right mt-1"
          right
          text
          v-for="(item, i) in button"
          :key="i"
          @click="
            () => {
              dismiss();
              if (item.event && typeof item.event === 'function') item.event();
            }
          "
        >
          {{ item.label }}
        </v-btn>
      </v-card-actions>
    </v-card>
  </v-snackbar>
</template>

<script>
export default {
  name: 'Toast',
  data() {
    return {
      active: false,
      text: '',
      icon: '',
      color: 'info',
      timeout: 4000,
      dismissible: true,
      button: [{ label: 'Tutup' }],
    };
  },
  methods: {
    reset() {
      this.text = '';
      this.icon = '';
      this.color = 'info';
      this.timeout = 4000;
      this.dismissible = true;
      this.button = [{ label: 'Tutup' }];
      this.event = null;
    },

    show(options = {}) {
      // reset
      this.reset();

      if (this.active) {
        this.close();
        this.$nextTick(() => this.show(options));
        return;
      }

      Object.keys(options).forEach(
        (field) =>
          (this[field] = options[field] !== false && options[field] !== undefined ? options[field] : this[field])
      );

      this.active = true;
    },

    close() {
      this.active = false;
    },

    dismiss() {
      if (this.dismissible) {
        this.active = false;
      }
    },
  },
};
</script>

<style scoped></style>
