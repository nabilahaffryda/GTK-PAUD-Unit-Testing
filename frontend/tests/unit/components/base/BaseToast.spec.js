import { mount, createLocalVue } from "@vue/test-utils";
import BaseToast from "@/components/base/BaseToast.vue";
import Vuetify from 'vuetify';
import Vue from 'vue';
import VueRouter from 'vue-router';

Vue.use(Vuetify)
const localVue = createLocalVue();
localVue.use(VueRouter)
const router = new VueRouter();

describe('BaseToast.vue', () => {
    let vuetify
    beforeEach(() => {
        vuetify = new Vuetify()
    })
    function wrapperFactory({ } = {}) {
        return mount(BaseToast, {
            localVue,
            vuetify,
            router,
        });
    }
    test('call reset method', () => {
        const wrapper = wrapperFactory();
        wrapper.vm.reset = jest.fn();
        wrapper.vm.reset();
        expect(wrapper.vm.reset.mock.calls.length).toBe(1);
    })
    test('call close method', () => {
        const wrapper = wrapperFactory();
        wrapper.vm.close = jest.fn();
        wrapper.vm.close();
        expect(wrapper.vm.close.mock.calls.length).toBe(1);
    })
    test('call show method', () => {
        const wrapper = wrapperFactory();
        wrapper.vm.show = jest.fn();
        wrapper.vm.show();
        expect(wrapper.vm.show.mock.calls.length).toBe(1);
    })
    test('call dismiss method', () => {
        const wrapper = wrapperFactory();
        wrapper.vm.dismiss = jest.fn();
        wrapper.vm.dismiss();
        expect(wrapper.vm.dismiss.mock.calls.length).toBe(1);
    })
})
