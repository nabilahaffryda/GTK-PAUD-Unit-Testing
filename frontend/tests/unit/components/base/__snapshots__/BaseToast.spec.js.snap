// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseToast.vue render toast 1`] = `
<div class="v-snack v-snack--has-background v-snack--right v-snack--top" style="padding-bottom: 0px; padding-top: 0px; opacity: 0.8;">
  <transition-stub name="v-snack-transition">
    <div class="v-snack__wrapper v-sheet theme--dark info" style="display: none;">
      <div role="status" aria-live="polite" class="v-snack__content">
        <div class="v-card v-card--flat v-sheet theme--dark transparent">
          <div class="v-card__text pa-0">
            <div role="list" class="v-list pa-0 v-sheet theme--dark transparent">
              <div tabindex="-1" role="listitem" class="v-list-item theme--dark">
                <!---->
                <div class="v-list-item__content"><span class="white--text body-1"></span></div>
              </div>
            </div>
          </div>
          <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
          <div class="v-card__actions">
            <div class="spacer"></div> <button type="button" class="text-right mt-1 v-btn v-btn--right v-btn--text theme--dark v-size--default"><span class="v-btn__content">
        Tutup
      </span></button>
          </div>
        </div>
      </div>
      <div class="v-snack__action "></div>
    </div>
  </transition-stub>
</div>
`;
